### ISSUE TITLE
Some items are not stacking

### ISSUE THREAD

**Issue by MiguelDP4** [HydraheadHunter/Data-Driven-Stacks#13]
After some time playing, some blocks stop stacking for no reason. Everything was going well until I made an afk session and got a bunch of bones. I wanted to make them bone blocks but couldn't stack them over 64 blocks. This is also happening with dried kelp blocks and regular dried kelp, they stop stacking over 64 after a while.

https://github.com/user-attachments/assets/bfa1b072-8f86-4c17-a7be-13ea5be238ff

https://github.com/user-attachments/assets/3fac3b42-c97a-494b-9ab7-ebef027f5ce6

All of the blocks are under the max_stack_size_2048.json file. I didn't put any of these items under players_stack_less.json, players_stack_more.json or villagers_stack_less.json files. 

Oh I forgot to share the datapack I'm using:
[DataDrivenStackData.zip](https://github.com/user-attachments/files/17247700/DataDrivenStackData.zip)

I'm playing on a server and I've lost some full stacks of items (2048 items to the void haha). It has only happened to me because I'm testing the mod but I wouldn't want my friends to have this happen to them.

**Comment by HydraheadHunter** [comment]
Yep, that do be a serious problem and I'll look into the cause and try to patch a solution: Thank you for bring it to my attention. 

I have an immediate plausible explanation that the behavior might be caused by some as-yet unknown mod and/or datapack incompatability. I'd like to disprove before digging back into the weeds of my code to figure out what I was thinking three months ago when I last touched it. Would you be willing and able to share what other mods and datapacks you have installed on this world so I can investigate and rule that out first.

Furthermore, assuming it's not an incompatability issue, would you also be willing and able to share your `playerdata/[uuid].dat` file with at least one bugged stack in your inventory, as it might assist in finding the cause of the error.

Again, thank you very much for bring this to my attention.

**Comment by MiguelDP4** [comment]
Sorry I didn't see your comment, I hope you get this message.

Here's the server side mod list:
ACE_mc1.20.6-3.1.1.jar
AnvilNeverTooExpensive-fabric-1.21-1.2.jar
appleskin-fabric-mc1.21-3.0.2.jar
armored-elytra-1.0.1.jar
balm-fabric-1.21-21.0.7.jar
betterspawnercontrol-1.21.0-4.5.jar
BetterThanMending-fabric-2.1.0.jar
chat_heads-0.12.3-fabric-1.21.jar
Chunky-1.4.16.jar
collective-1.21.0-7.62.jar
convenient-mobgriefing-2.1.2.jar
datastacks-0.9.5.jar
dragondropselytra-1.21.0-3.3.jar
erodingstoneentities-1.21.0-4.5.jar
fabric-api-0.100.3+1.21.jar
FallingTree-1.21-1.21.0.2.jar
ferritecore-7.0.0-fabric.jar
forgivingvoid-fabric-1.21-21.0.2.jar
furnacerecycle-1.21.0-2.4.jar
glowing-torchflower-mc1.20.6-1.0.0.jar
grassseeds-1.21.0-3.2.jar
healingcampfire-1.21.0-5.4.jar
hoetweaks-1.21.0-3.4.jar
infinitetrading-1.21.0-4.4.jar
inventorytotem-1.21.0-3.2.jar
justmobheads-1.21.0-7.9.jar
justplayerheads-1.21.0-4.0.jar
lithium-fabric-mc1.21-0.12.7.jar
lmft-1.0.4+1.21-fabric.jar
morezombievillagers-1.21.0-3.5.jar
nametagtweaks-1.21.0-3.6.jar
noanimaltemptdelay-1.21.0-1.1.jar
PerfectAccuracy-1.3.jar
quicksort-0.10.0+1.21.jar
realisticbees-1.21.0-3.9.jar
shulkerbox-fabric-1.21-2.0.5.jar
shulkerdropstwo-1.21.0-3.3.jar
smallernetherportals-1.21.0-3.7.jar
wooltweaks-1.21.0-3.6.jar
zombiehorsespawn-1.21.0-5.0.jar
Zombify-0.1.4.jar

Here's the mods folder if you need it:
[mods.zip](https://github.com/user-attachments/files/17346827/mods.zip)

The only mod that deals with inventories other than yours is quicksort. 

Here's the playerdata file. Specifically mine, as I'm the one playing in the video. I'm not worried about the current losses as I'm testing the mod right now and I haven't lost anything too rare, valuable or irreplaceable. Besides, I'm the server admin and I could just /give the lost items.
[3a3efb44-8b59-4504-b830-f641c7c434d5.zip](https://github.com/user-attachments/files/17346802/3a3efb44-8b59-4504-b830-f641c7c434d5.zip)

Here are the datapacks, most of them are just recipe changes.
[datapacks.zip](https://github.com/user-attachments/files/17346857/datapacks.zip)


Anyway,
I'm a programmer. I work in web development so I have no experience in minecraft modding but if you are willing to point me in the right direction I would be interested to learn and help you, free of charge of course.

**Comment by MiguelDP4** [comment]
There's one more detail I forgot to mention. It seems when I log out of the server and log in again, I'm able to stack up to the configured values again. It seems like the server forgets the initial configuration after some time and stops setting the tag on the items.

**Comment by HydraheadHunter** [comment]
Thank you for this information: it's much appreciated. Your datapacks definitely ain't causing compatability issues and neither are most of the mods. The only ones that might be interacting badly are LMFT and ferritecore but if there's a bad interaction there, its definitely a `my mod's the one causing the problem` sort of thing.

Looking at your player.dat, am I correct in assuming that the bamboo was the bugged item? Because all the other items that should have a max_stack_size tag have `max_stack_size:2048`.

**Comment by MiguelDP4** [comment]
The items giving me issues are the dried kelp blocks and the dried kelp, but most of the items over 64 forget the max_stack_size attribute after a while. 

For example, take a 2048 stack of dried kelp and craft it into blocks. Instead of making a big stack of dried kelp blocks, you get a bunch of stacks of 64. The same happens with bone blocks and redstone blocks.
